hs/Database/Tables.hs

```diff
@@ -44,6 +44,11 @@ derivePersistField "Time"
 type Point = (Double, Double)
 
 share [mkPersist sqlSettings, mkMigrate "migrateAll"] [persistLowerCase|
+
+Department json
+    code T.Text
```

Comment:
This can be a list of codes (`[T.Text]`) for now, since there can be multiple codes per department. In your function above, you'll need to group the codes together by department.